docs/DeveloperGuide.adoc

```diff
@@ -774,7 +1074,25 @@ Use case resumes at step 2.
 
 .  Should work on any link:#mainstream-os[mainstream OS] as long as it has Java `1.8.0_60` or higher installed.
 .  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
+.  Should be compatible with Mac and Windows users
+.  Should be able to handle unicode (UTF-8) text
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+.  The system should be backward compatible with data produced by earlier versions of the system.
+.  The system should work on both 32-bit and 64-bit environments.
+.  Should return the list of persons in the storage within 1 second when users sort persons by tag.
+.  Should be able to recover itself after a power shutdown.
+.  Should display specific error messages if the system fails react to user.
+.  Should be able to give suggestions to user on how to handle an error messages.
+.  This product should be free for the public.
+.  Should be easy for user to remember the commands.
+.  Should be backward compatible with data produced by earlier versions of the system.
+.  Should be usable by a novice who has never tried command line interface.
+.  The project is expected to adhere to a schedule that delivers a feature set every week.
+.  The product is not required to handle the printing of reports.
+.  Should produce clean code when contribute to the project.
```

Comment:
??

Reply:
Oh. I mean the when developers contribute to the project, the style has to follow "clean code" style.